lib/ecto/adapters/clickhouse/structure.ex

```diff
@@ -94,4 +83,34 @@ defmodule Ecto.Adapters.ClickHouse.Structure do
       {:error, reason, _conn} -> {:error, reason}
     end
   end
+
+  defp run_with_cmd(cmd, opts, opt_args, cmd_opts \\ []) do
+    unless System.find_executable(cmd) do
+      raise "could not find executable `#{cmd}` in path, " <>
+              "please guarantee it is available before running ecto commands"
+    end
+
+    args = []
+
+    args = if username = opts[:username], do: ["--username", username | args], else: args
+
+    args = if password = opts[:password], do: ["--password", password | args], else: args
+
+    args = if port = opts[:port], do: ["--port", to_string(port) | args], else: args
+    args = if database = opts[:database], do: ["--database", database | args], else: args
+
+    host = opts[:hostname] || "localhost"
+
+    args = ["--host", host | args]
+    args = args ++ opt_args
+
+    cmd_opts =
+      cmd_opts
+      |> Keyword.put_new(:stderr_to_stdout, true)
+
+    case System.cmd(cmd, args, cmd_opts) do
+      {output, 0} -> {:ok, output}
+      {output, 1} -> {:error, output}
```

Comment:
It is always status code 1 for clickhouse-client errors?

Reply:
I don't know if it always will be the case but I tried locally and it returned 1 for me on errors

Reply:
Maybe we can do `{output, _}` just in case it doesn't. Because otherwise the error message would be formatted badly and possibly misleading (MatchError)

Reply:
done

Reply:
It seems like `run_with_cmd` is not returning a result (ok / error) tuples in Postgres but rather `{status, output}`

Reply:
https://hexdocs.pm/ecto_sql/Ecto.Adapter.Structure.html#c:dump_cmd/3 requires `{output, status}` returned. To make it easier to implement it, can you please make `run_with_cmd` return `{status, output}`?

Reply:
sure

Reply:
done